### DISCUSSION TITLE
Arbitrary file picker

### DISCUSSION THREAD

**Discussion by mid-kid** [lxqt/lxqt#2303] [Ideas]
I'm looking for a way to use the xdg-desktop-portal to call arbitrary programs to help me pick a file. Ideally I'd be able to write a shell script presenting me a dialog with multiple options, one of which would for example be nsxiv -o, which returns a list of files picked, allowing me to pick images easily.

Seeing as how this portal essentially only implements file picking so far, doing so in very little code, I think it wouldn't be too hard for me to modify it to do what I want, but I'm wondering if this is something you might be interested in having, or whether I should just fork off.

If this isn't something that's possible due to the design of the D-Bus portal specification be sure to tell me too, I've done very little research yet, this has mostly been shower thoughts so far.

**Comment by tsujan** [comment]
Thanks for using Discussions!

I think your question/idea is beyond LXQt. `xdg-desktop-portal-lxqt` is LXQt's backend implementation for `xdg-desktop-portal` — e.g., for using LXQt file dialog with apps like Firefox — and should only support LXQt file dialog.

**Reply by mid-kid** [reply]
I see, that's a shame. Thanks for answering.
